ivy/functional/ivy/device.py

```diff
@@ -91,6 +92,7 @@ def __enter__(self):
         "cpu"
         """
         ivy.set_default_device(self._dev)
+        ivy.set_soft_device_mode(True)
```

Comment:
Probably you've mentioned this already, but not sure why we're enabling the `soft_device_mode` inside `default_device` if the default value of `soft_device_mode` seems to be [`False`](https://github.com/unifyai/ivy/pull/17623/files#diff-ca7db18115e5b52f90e231d41fae2418e12b8598b43cb7bb23ab9448d4d13af8R342)

Reply:
The change was made so as to use `ivy.DefaultDevice` in the same way as `tf.device` context manager. Every operation in the context should be performed in the same device provided to the context manager. For this to happen we set soft device mode to `True` and unset it(in `__exit__`) :slightly_smiling_face:

Reply:
Oh okay, makes sense! Thanks for clarifying 👍